src/main/java/bozo/Storage.java

```diff
@@ -0,0 +1,93 @@
+package bozo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the storage of the tasks.
+ */
+public class Storage {
+    private final String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the tasks from the file.
+     *
+     * @return The list of tasks.
+     */
+    public ArrayList<Task> loadFile() {
+        ArrayList<Task> list = new ArrayList<>();
+        try {
+            File txtFile = new File(filePath);
+            Scanner sc = new Scanner(txtFile);
+            while (sc.hasNextLine()) {
+                String line = sc.nextLine();
+                String[] parts = line.split(" \\| ");
+                String type = parts[0];
+                int i = Integer.parseInt(parts[1]);
+                String s = parts[2];
+
+                switch (type) {
+                case "T":
+                    Todo td = new Todo(s);
+                    if (i == 1) {
+                        td.isDone = true;
+                    } else {
+                        td.isDone = false;
+                    }
+                    list.add(td);
+                    break;
+                case "D":
+                    Deadline d = new Deadline(s, parts[3]);
+                    if (i == 1) {
+                        d.isDone = true;
+                    } else {
+                        d.isDone = false;
+                    }
+                    list.add(d);
+                    break;
+                case "E":
+                    Event e = new Event(s, parts[3], parts[4]);
+                    if (i == 1) {
+                        e.isDone = true;
+                    } else {
+                        e.isDone = false;
+                    }
+                    list.add(e);
+                    break;
+                }
```

Comment:
i think the if else statement can be moved down to after the switch statement and assign the todo, deadline, event object to Task, then do one if else on that task.